aws-sns-topicinlinepolicy/src/main/java/software/amazon/sns/topicinlinepolicy/ReadHandler.java

```diff
@@ -0,0 +1,51 @@
+package software.amazon.sns.topicinlinepolicy;
+
+import com.amazonaws.util.StringUtils;
+import software.amazon.awssdk.services.sns.SnsClient;
+import software.amazon.awssdk.services.sns.model.GetTopicAttributesResponse;
+import software.amazon.awssdk.utils.CollectionUtils;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.HandlerErrorCode;
+
+import java.util.Map;
+
+public class ReadHandler extends BaseHandlerStd {
+    private Logger logger;
+
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final ProxyClient<SnsClient> proxyClient,
+            final Logger logger) {
+        this.logger = logger;
+
+
+        final ResourceModel model = request.getDesiredResourceState();
+
+        if (model == null || StringUtils.isNullOrEmpty(model.getTopicArn())) {
+            return ProgressEvent.failed(model, callbackContext, HandlerErrorCode.InvalidRequest, EMPTY_POLICY_AND_TOPICARN_ERROR_MESSAGE );
+        }
+
+        logger.log(String.format("StackId: %s, ClientRequestToken: %s] Calling Read TopicInlinePolicy", request.getStackId(), request.getClientRequestToken()));
+        if (!doesTopicPolicyExist(proxyClient, request, model)) {
```

Comment:
does the native code call the GetAPI in your CreateHandler? if is does not you will need to implement a soft failure for adding a read permission inside your create handler. If you need an example please dm me